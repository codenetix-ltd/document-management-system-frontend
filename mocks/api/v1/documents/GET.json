{
  "links": {
    "first": "/api/v1/documents?page=1",
    "last": "/api/v1/documents?page=4",
    "prev": null,
    "next": "/api/v1/documents?page=2"
  },
  "meta": {
    "currentPage": 1,
    "from": 1,
    "lastPage": 8,
    "path": "/api/v1/documents",
    "perPage": 15,
    "to": 15,
    "total": 51
  },
  "data": [
    {
      "name": "Wooden sticks Magnum",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 1,
      "templateId": 1,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Wooden sticks Classic",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 2,
      "templateId": 2,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Stick straight 114 mm",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 3,
      "templateId": 3,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Birchwood sticks 114x10x2 mm",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 4,
      "templateId": 4,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Sticks Butterfly 93 mm RPK",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 5,
      "templateId": 5,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Sticks Butterfly 93 mm RPK",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 6,
      "templateId": 6,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Wooden sticks Mag",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 7,
      "templateId": 1,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Wooden sticks Class",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 8,
      "templateId": 2,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Stick straight",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 9,
      "templateId": 3,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Birchwood sticks",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 10,
      "templateId": 4,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Sticks Butterfly",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 11,
      "templateId": 5,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Wooden sticks",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 12,
      "templateId": 2,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Stick straight new",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 13,
      "templateId": 3,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Birchwood sticks new",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 14,
      "templateId": 4,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    },
    {
      "name": "Sticks Butterfly new",
      "ownerId": 1,
      "substituteDocumentId": 1,
      "attributesValues": [
        {
          "id": 1,
          "value": "Example value"
        }
      ],
      "id": 15,
      "templateId": 5,
      "createdAt": "1503224430",
      "updatedAt": "1503310830"
    }
  ]
}
