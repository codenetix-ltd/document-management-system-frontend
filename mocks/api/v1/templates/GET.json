{
  "links": {
    "first": "http://localhost/api/v1/users?page=1",
    "last": "http://localhost/api/v1/users?page=4",
    "prev": null,
    "next": "http://localhost/api/v1/users?page=2"
  },
  "meta": {
    "currentPage": 1,
    "from": 1,
    "lastPage": 4,
    "path": "http://localhost/api/v1/users",
    "perPage": 15,
    "to": 15,
    "total": 51
  },
  "data": [
    { "id": 1, "name": "Bibendum morbi" },
    { "id": 2, "name": "Morbi" },
    { "id": 3, "name": "Posuere cubilia" },
    { "id": 4, "name": "Lorem quisque" },
    { "id": 5, "name": "Fusce" },
    { "id": 6, "name": "Ultrices posuere" },
    { "id": 7, "name": "Mattis odio donec" },
    { "id": 8, "name": "Suspendisse potenti" },
    { "id": 9, "name": "Lectus suspendisse potenti" },
    { "id": 10, "name": "Tellus in" },
    { "id": 11, "name": "A ipsum integer" },
    { "id": 12, "name": "Ut nulla" },
    { "id": 13, "name": "Donec" },
    { "id": 14, "name": "Lectus" },
    { "id": 15, "name": "A feugiat" }
  ]
}
