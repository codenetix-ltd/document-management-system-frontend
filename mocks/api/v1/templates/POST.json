{
  "name": "Example template",
  "id": 1
}
