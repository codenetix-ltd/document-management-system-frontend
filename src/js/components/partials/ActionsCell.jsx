import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { DataLink } from 'Components/common/dataControls';

/**
 * Component for react-table's actions column with edit and delete buttons
 * @param props
 * @returns {*}
 * @constructor
 */
export default function ActionsCell(props) {
  const { editLink, rowData, onDelete } = props;
  return (
    <div>
      <Link to={editLink} className="btn-success btn btn-xs">
        <i className="fa fa-edit" />
      </Link>
      &nbsp;
      <DataLink cls="btn-danger btn btn-xs" data={rowData} onClick={onDelete}>
        <i className="fa fa-trash" />
      </DataLink>
    </div>
  );
}

ActionsCell.propTypes = {
  editLink: PropTypes.string.isRequired,
  rowData: PropTypes.any.isRequired,
  onDelete: PropTypes.func.isRequired
};
