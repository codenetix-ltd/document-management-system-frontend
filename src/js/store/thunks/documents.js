import { API } from 'Config';
import pick from 'lodash/pick';
import pickBy from 'lodash/pickBy';
import identity from 'lodash/identity';

import axios from 'Services/request';

import {
  $loading,
  $document,
  $updateDocument,
  $documentsList,
  $setComparedDocuments
} from 'Store/actions';

import { $$messageSet } from 'Store/thunks/message';
import { $$errorSet } from 'Store/thunks/error';

import initialState from 'Store/reducers/initialState.json';

const fn = () => {};

export const $$documentsFetch = (dispatch, params, callback = fn) => {
  const picked = pick(params, ['page', 'sortField', 'sortDirection']);
  const filters = pickBy(params.filterSet, identity);
  dispatch($loading(true));
  axios.get(API.documents, {
    params: { ...picked, ...filters }
  }).then(({ data }) => {
    dispatch($documentsList({
      list: data.data,
      lastPage: data.meta.lastPage,
      ...picked
    }));
    dispatch($loading(false));
    callback(data);
  }).catch(err => {
    console.trace(err);
    dispatch($loading(false));
    $$errorSet(dispatch, err);
  });
};

export const $$documentFetch = (dispatch, documentID, callback = fn) => {
  dispatch($loading(true));
  axios.get(`${API.documents}/${documentID}`).then(({ data }) => {
    dispatch($document({ ...data }));
    dispatch($loading(false));
    callback(data);
  }).catch(err => {
    console.trace(err);
    dispatch($loading(false));
    $$errorSet(dispatch, err);
  });
};

export const $$documentUpdate = (dispatch, part) => {
  dispatch($updateDocument(part));
};

export const $$documentReset = (dispatch) => {
  dispatch($document(initialState.document));
};

export const $$documentDelete = (dispatch, { id, name }, close = fn) => {
  axios.delete(`${API.documents}/${id}`).then(() => {
    $$documentsFetch(dispatch, 1);
    $$messageSet(dispatch, {
      type: 'success',
      text: `The document ${name} was successfully deleted.`
    });
    close();
  }).catch(err => {
    console.trace(err);
    close();
    $$errorSet(dispatch, err);
  });
};

export const $$documentsMassDelete = (dispatch, ids, close = fn) => {
  const p = Promise.all(ids.map(id => axios.delete(`${API.documents}/${id}`)));
  p.then(() => {
    $$documentsFetch(dispatch, 1);
    $$messageSet(dispatch, {
      type: 'success',
      text: 'Selected documents were successfully deleted.'
    });
    close();
  }).catch(err => {
    console.trace(err);
    close();
    $$errorSet(dispatch, err);
  });
};

export const $$comparedDocumentsFetch = (dispatch, ids, callback = fn) => {
  const p = Promise.all(ids.map(id => axios.get(`${API.documents}/${id}`)));
  p.then(d => {
    dispatch($setComparedDocuments(d.map(({ data }) => data)));
    callback(d);
  }).catch(err => {
    console.trace(err);
  });
};
