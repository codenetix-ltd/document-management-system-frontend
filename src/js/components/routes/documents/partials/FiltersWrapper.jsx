import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import autobind from 'autobind-decorator';

import InputFilter from 'Components/common/filters/Input';
import SelectFilter from 'Components/common/filters/Select';
import DatePicker from 'Components/common/filters/DatePicker';
import CheckboxFilter from 'Components/common/filters/Checkbox';

import { $documentsFilter } from 'Store/actions';
import { $$documentsFetch } from 'Store/thunks/documents';
import { $$templatesFetch } from 'Store/thunks/templates';
import { $$labelsFetch } from 'Store/thunks/labels';

@autobind
export class FiltersWrapper extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    documents: PropTypes.any.isRequired,
    templates: PropTypes.any.isRequired,
    labels: PropTypes.any.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      templateOptions: [],
      labelOptions: []
    };
  }

  componentDidMount() {
    $$templatesFetch(this.props.dispatch, 1, () => {
      $$labelsFetch(this.props.dispatch, 1);
    });
  }

  componentWillReceiveProps(nextProps) {
    const { list: templatesList } = nextProps.templates;
    const { list: labelsList } = nextProps.labels;
    if (templatesList.length) {
      this.setState({
        templateOptions: templatesList.map(({ id, name }) => ({
          value: id, label: name
        }))
      });
    }
    if (labelsList.length) {
      this.setState({
        labelOptions: labelsList.map(({ id, name }) => ({
          value: id, label: name
        }))
      });
    }
  }

  onFilterChange(filterType, value) {
    const { dispatch, documents } = this.props;
    const {
      page,
      orderBy,
      sortedBy
    } = documents;
    if (Array.isArray(value)) {
      value = value.map(item => item.value).join(','); // eslint-disable-line
    }
    dispatch($documentsFilter({
      [`filters[${filterType}]`]: value
    }));
    setTimeout(() => {
      const { documents: docs } = this.props;
      $$documentsFetch(dispatch, {
        page, orderBy, sortedBy, filterSet: docs.filterSet
      });
    }, 1);
  }

  render() {
    const { templateOptions, labelOptions } = this.state;
    return (
      <div className="document-filters-wrapper datatables-filters-wrapper">
        <div className="document-filters datatables-filters">
          <div className="row">

            <div className="form-group col-lg-1 col-md-4 col-sm-6">
              <InputFilter label="ID" filterType="id" placeholder="Id" onChange={this.onFilterChange} />
            </div>

            <div className="form-group col-lg-3 col-md-4 col-sm-6">
              <InputFilter label="Name" filterType="name" placeholder="Document name" onChange={this.onFilterChange} />
            </div>

            <div className="form-group col-lg-2 col-md-4 col-sm-6">
              <InputFilter label="Owner" filterType="ownerId" placeholder="Username" onChange={this.onFilterChange} />
            </div>

            <div className="form-group col-lg-2 col-md-4 col-sm-6">
              <SelectFilter
                label="Template"
                filterType="templateIds"
                options={templateOptions}
                onChange={this.onFilterChange}
              />
            </div>

            <div className="form-group col-lg-2 col-md-4 col-sm-6">
              <SelectFilter
                label="Labels"
                filterType="labelIds"
                options={labelOptions}
                onChange={this.onFilterChange}
              />
            </div>

            <div className="form-group col-lg-3 col-md-6 col-sm-6">
              <DatePicker
                label="Created at"
                filterType="createdAt.from"
                placeholder="Date from"
                onChange={this.onFilterChange}
              />
            </div>

            <div className="form-group col-lg-2 col-md-6 col-sm-6">
              <DatePicker
                filterType="createdAt.to"
                placeholder="Date to"
                onChange={this.onFilterChange}
              />
            </div>

            <div className="form-group col-lg-3 col-md-6 col-sm-6">
              <DatePicker
                label="Updated at"
                filterType="updatedAt.from"
                placeholder="Date from"
                onChange={this.onFilterChange}
              />
            </div>

            <div className="form-group col-lg-2 col-md-6 col-sm-6">
              <DatePicker
                filterType="updatedAt.to"
                placeholder="Date to"
                onChange={this.onFilterChange}
              />
            </div>

            <div className="form-group col-lg-1 col-md-2 col-sm-2">
              <CheckboxFilter
                label="Archived"
                onChange={this.onFilterChange}
                filterType="archived"
              />
            </div>

          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ documents, templates, labels }) => ({ documents, templates, labels });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(FiltersWrapper);
