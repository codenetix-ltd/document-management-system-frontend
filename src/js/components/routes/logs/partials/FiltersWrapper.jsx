import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import autobind from 'autobind-decorator';

import InputFilter from 'Components/common/filters/Input';
import SelectFilter from 'Components/common/filters/Select';
import DatePicker from 'Components/common/filters/DatePicker';

import { $logsFilter } from 'Store/actions';
import { $$logsFetch } from 'Store/thunks/logs';

@autobind
export class FiltersWrapper extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    logs: PropTypes.any.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      typeOptions: []
    };
  }

  componentDidMount() {
    this.setState({
      typeOptions: [
        { value: 1, label: 'user' },
        { value: 2, label: 'document' },
        { value: 3, label: 'template' },
        { value: 4, label: 'label' }
      ]
    });
  }

  onFilterChange(filterType, value) {
    const { dispatch, logs } = this.props;
    const {
      page,
      orderBy,
      sortedBy
    } = logs;
    if (Array.isArray(value)) {
      value = value.map(item => item.value).join(','); // eslint-disable-line
    }
    dispatch($logsFilter({
      [`filter[${filterType}]`]: value
    }));
    setTimeout(() => {
      const { logs: _logs } = this.props;
      $$logsFetch(dispatch, {
        page, orderBy, sortedBy, filterSet: _logs.filterSet
      });
    }, 1);
  }

  render() {
    return (
      <div className="datatables-filters-wrapper">
        <div className="datatables-filters">
          <div className="row">
            <div className="form-group col-md-2">
              <div className="input-group">
                <InputFilter label="ID" onChange={this.onFilterChange} filterType="id" placeholder="Id" />
              </div>
            </div>
            <div className="form-group col-md-2">
              <div className="input-group">
                <InputFilter label="User" onChange={this.onFilterChange} filterType="user" placeholder="User" />
              </div>
            </div>
            <div className="form-group col-md-2">
              <div className="input-group">
                <InputFilter label="Action" onChange={this.onFilterChange} filterType="action" placeholder="Action" />
              </div>
            </div>
            <div className="form-group col-md-2">
              <div className="input-group input-group-sm">
                <SelectFilter
                  label="Type"
                  onChange={this.onFilterChange}
                  options={this.state.typeOptions}
                  filterType="type"
                />
              </div>
            </div>

            <div className="form-group col-md-2">
              <div className="input-group input-group-sm">
                <DatePicker
                  label="Created at"
                  onChange={this.onFilterChange}
                  filterType="createdAt.from"
                  placeholder="From"
                />
              </div>
            </div>
            <div className="form-group col-md-2">
              <div className="input-group input-group-sm">
                <DatePicker
                  onChange={this.onFilterChange}
                  filterType="createdAt.to"
                  placeholder="To"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ logs, templates }) => ({ logs, templates });

const mapDispatchToProps = (dispatch) => ({ dispatch });

export default connect(mapStateToProps, mapDispatchToProps)(FiltersWrapper);
