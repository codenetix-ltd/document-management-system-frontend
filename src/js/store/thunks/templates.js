import { API } from 'Config';
import axios from 'axios';

import {
  $loading,
  $templatesList,
  $template,
  $updateTemplate
} from 'Store/actions';

import { $$messageSet } from 'Store/thunks/message';
import { $$errorSet } from 'Store/thunks/error';

import initialState from 'Store/reducers/initialState.json';

const fn = () => {};

export const $$templatesFetch = (dispatch, page, callback = fn) => {
  dispatch($loading(true));
  axios.get(API.templates, {
    params: { page }
  }).then(({ data }) => {
    dispatch($templatesList({
      list: data.data,
      lastPage: data.meta.lastPage
    }));
    dispatch($loading(false));
    callback(data);
  }).catch(err => {
    console.trace(err);
    dispatch($loading(false));
    $$errorSet(dispatch, err);
  });
};

export const $$templateFetch = (dispatch, templateID, callback = fn) => {
  dispatch($loading(true));
  axios.get(`${API.templates}/${templateID}`).then(({ data }) => {
    dispatch($template({ ...data }));
    dispatch($loading(false));
    callback(data);
  }).catch(err => {
    console.trace(err);
    dispatch($loading(false));
    $$errorSet(dispatch, err);
  });
};

export const $$templateUpdate = (dispatch, part) => {
  dispatch($updateTemplate(part));
};

export const $$templateReset = (dispatch) => {
  dispatch($template(initialState.template));
};

export const $$templateDelete = (dispatch, { id, name }, close = fn) => {
  axios.delete(`${API.templates}/${id}`).then(() => {
    $$templatesFetch(dispatch, 1);
    $$messageSet(dispatch, {
      type: 'success',
      text: `The template ${name} was successfully deleted.`
    });
    close();
  }).catch(err => {
    console.trace(err);
    close();
    $$errorSet(dispatch, err);
  });
};
