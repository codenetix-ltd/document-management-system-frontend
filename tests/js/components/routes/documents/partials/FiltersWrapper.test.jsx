import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { FiltersWrapper } from 'Routes/documents/partials/FiltersWrapper';

import { $$templatesFetch } from 'Store/thunks/templates';
import { $$documentsFetch } from 'Store/thunks/documents';

jest.mock('Store/thunks/templates', () => ({
  $$templatesFetch: jest.fn()
}));

jest.mock('Store/thunks/documents', () => ({
  $$documentsFetch: jest.fn()
}));

describe('Documents FiltersWrapper', () => {
  let wrapper;
  let props;
  let instance;
  let dispatch;

  beforeEach(() => {
    dispatch = jest.fn();
    props = {
      dispatch,
      documents: {
        list: []
      },
      templates: {
        list: []
      },
      labels: {
        list: []
      }
    };
    wrapper = shallow(<FiltersWrapper {...props} />);
    instance = wrapper.instance();
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should call $$templatesFetch on componentDidMount', () => {
    expect($$templatesFetch.mock.calls.length).toBe(1);
  });

  it('should change templateOptions state on componentWillReceiveProps', () => {
    const nextProps = {
      templates: {
        list: [
          { id: 1, name: 'Lorem' },
          { id: 2, name: 'Ipsum' }
        ]
      }
    };
    expect(instance.state).toHaveProperty('templateOptions');
    expect(instance.state.templateOptions).toBeInstanceOf(Array);
    expect(instance.state.templateOptions.length).toBe(0);
    wrapper.setProps(nextProps, () => {
      expect(instance.state.templateOptions.length).toBe(2);
    });
  });

  it('should change labelOptions state on componentWillReceiveProps', () => {
    const nextProps = {
      labels: {
        list: [
          { id: 1, name: 'Lorem' },
          { id: 2, name: 'Ipsum' }
        ]
      }
    };
    expect(instance.state).toHaveProperty('labelOptions');
    expect(instance.state.labelOptions).toBeInstanceOf(Array);
    expect(instance.state.labelOptions.length).toBe(0);
    wrapper.setProps(nextProps, () => {
      expect(instance.state.labelOptions.length).toBe(2);
    });
  });

  it('should call dispatch when calling onFilterChange', () => {
    expect(dispatch).not.toBeCalled();
    instance.onFilterChange('name', 'Lorem ipsum');
    expect(dispatch).toHaveBeenCalledTimes(1);
  });

  it('should call $$documentsFetch when calling onFilterChange', () => {
    expect($$documentsFetch).not.toBeCalled();
    instance.onFilterChange('name', 'Lorem ipsum');
    setTimeout(() => {
      expect($$documentsFetch).toHaveBeenCalledTimes(1);
    }, 100);
  });
});
