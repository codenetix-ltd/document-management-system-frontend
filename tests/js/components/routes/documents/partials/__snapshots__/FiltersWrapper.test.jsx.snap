// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documents FiltersWrapper should render correctly 1`] = `
<div
  className="document-filters-wrapper datatables-filters-wrapper"
>
  <div
    className="document-filters datatables-filters"
  >
    <div
      className="row"
    >
      <div
        className="form-group col-lg-1 col-md-4 col-sm-6"
      >
        <InputFilter
          filterType="id"
          label="ID"
          onChange={[Function]}
          placeholder="Id"
        />
      </div>
      <div
        className="form-group col-lg-3 col-md-4 col-sm-6"
      >
        <InputFilter
          filterType="name"
          label="Name"
          onChange={[Function]}
          placeholder="Document name"
        />
      </div>
      <div
        className="form-group col-lg-2 col-md-4 col-sm-6"
      >
        <InputFilter
          filterType="owner"
          label="Owner"
          onChange={[Function]}
          placeholder="Username"
        />
      </div>
      <div
        className="form-group col-lg-2 col-md-4 col-sm-6"
      >
        <SelectFilter
          filterType="template"
          label="Template"
          onChange={[Function]}
          options={Array []}
        />
      </div>
      <div
        className="form-group col-lg-2 col-md-4 col-sm-6"
      >
        <SelectFilter
          filterType="label"
          label="Labels"
          onChange={[Function]}
          options={Array []}
        />
      </div>
      <div
        className="form-group col-lg-3 col-md-6 col-sm-6"
      >
        <DatePickerFilter
          filterType="createdAt.from"
          label="Created at"
          onChange={[Function]}
          placeholder="Date from"
        />
      </div>
      <div
        className="form-group col-lg-2 col-md-6 col-sm-6"
      >
        <DatePickerFilter
          filterType="createdAt.to"
          label=""
          onChange={[Function]}
          placeholder="Date to"
        />
      </div>
      <div
        className="form-group col-lg-3 col-md-6 col-sm-6"
      >
        <DatePickerFilter
          filterType="updatedAt.from"
          label="Updated at"
          onChange={[Function]}
          placeholder="Date from"
        />
      </div>
      <div
        className="form-group col-lg-2 col-md-6 col-sm-6"
      >
        <DatePickerFilter
          filterType="updatedAt.to"
          label=""
          onChange={[Function]}
          placeholder="Date to"
        />
      </div>
      <div
        className="form-group col-lg-1 col-md-2 col-sm-2"
      >
        <CheckboxFilter
          filterType="archived"
          label="Archived"
          onChange={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
